### PR TITLE
Fix Issue 23669 - [DIP1000] Compound assignment to length of slice me…

### PR DESCRIPTION
**Pull request by ntrel**
…mber variable in scope method fails

Use `ref` instead of pointer.

### CODE REVIEW

File: compiler/test/runnable/safearrgrow.d [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    {
+        a.length += 1;
+    }
+}
+
+void main()
+{
+    S s;
+    s.reserve;
+    assert(s.a.length >= 1);
```
Comment (dkorpel) [inline compiler/test/runnable/safearrgrow.d:15]: This test seems unrelated to the issue. I'd make this a compilable test, runnable tests are slower.

**Comment by ntrel** [comment]
Ubuntu tests failing:
```
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
```

**Review by dkorpel — APPROVED** [review]
